src/components/OperatorChecklist.tsx

```diff
@@ -0,0 +1,289 @@
+            </ChecklistItem>
+            <Separator />
+            <ChecklistItem
+                state={nodesDeclared}
+                tip={
+                    <p>
+                        Configure your nodes with private keys and declare the
+                        corresponding addresses to the operator contract.
+                    </p>
+                }
+            >
+                Node addresses declared
+            </ChecklistItem>
+            <Separator />
+            <ChecklistItem
+                state={nodesFunded}
+                tip={
+                    <p>
+                        Your node addresses must be funded with some MATIC for Polygon
+                        transaction fees. This alert triggers if a balance is less than
+                        0.1 MATIC.
+                    </p>
+                }
+            >
+                Node addresses funded
+            </ChecklistItem>
+            <Separator />
+            <ChecklistItem
+                state={nodesRunning}
+                tip={
+                    <p>
+                        You need to run Streamr nodes with the operator plugin enabled.
+                        The nodes will do the work you sign up for when you stake on
+                        sponsorships.
+                    </p>
+                }
+            >
+                Nodes running
+            </ChecklistItem>
+            <Separator />
+            <ChecklistItem
+                disabled
+                state={nodesReachable}
+                tip={<p>The websocket port on your nodes should be reachable.</p>}
+            >
+                Nodes reachable
+            </ChecklistItem>
+        </>
+    )
+}
+
+interface OperatorChecklist {
+    funded?: boolean | undefined
+    nodesDeclared?: boolean | undefined
+    nodesFunded?: boolean | undefined
+    nodesReachable?: boolean | undefined
+    nodesRunning?: boolean | undefined
+}
+
+function useOperatorChecklist(operatorId: string | undefined): OperatorChecklist {
+    const operatorQuery = useOperatorByIdQuery(operatorId)
+
+    const { data: operator, isLoading, isFetching } = operatorQuery
+
+    const [nodesFunded, setNodesFunded] = useState<boolean>()
+
+    const [nodesReachable] = useState<boolean>(false)
+
+    const heartbeats = useInterceptHeartbeats(operatorId)
+
+    const { count, isLoading: isLoadingLiveNodes } = useOperatorLiveNodes(heartbeats)
+
+    const nodesRunning = isLoadingLiveNodes ? undefined : count > 0
+
+    useEffect(() => {
+        setNodesFunded(undefined)
+
+        if (!operator) {
+            return
+        }
+
+        if (!operator.nodes.length) {
+            /**
+             * No nodes = no funded nodes, eh? Let's report that
+             * and cut it short!
+             */
+            return void setNodesFunded(false)
+        }
+
+        let mounted = true
+
+        setTimeout(async () => {
+            class InsufficientFundsError {}
```

Comment:
wait, what? why do you define a class inside a timeout? it should be defined outside of the hook